main.js

```diff
@@ -437,7 +458,6 @@
         console.log('All data loaded');
     }).catch((err) => {
         console.log('final error catch data', err);
```

Comment:
The simplest way might be to wrap this chunk that calls `Promise.all` in a function--something like `fetchAllData`. Then on the "retry" button click handler we could just call `fetchAllData` again.